torch_xla/csrc/runtime/pjrt_registry.cc

```diff
@@ -98,9 +98,10 @@ InitializePjRt(const std::string& device_type) {
 
     xla::PjRtClient::KeyValueGetCallback kv_get = nullptr;
     xla::PjRtClient::KeyValuePutCallback kv_put = nullptr;
-    auto allowed_devices =
-        std::make_optional<std::set<int>>(std::set{local_process_rank});
+    XLA_CHECK(global_world_size>=0) << "global_world_size must be non-negative. LOCAL_WORLD_SIZE=" << local_world_size << ", WORLD_SIZE=" << global_world_size;
+    std::optional<std::set<int>> allowed_devices;
     if (global_world_size > 1) {
+      allowed_devices = std::set{local_process_rank};
```

Comment:
Where does `allowed_devices` get used?

Reply:
It is used in `xla::GetStreamExecutorGpuClient` when torch_xla creates the StreamExecutorGpuClient